uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/OrchestratorClaimAT.java

```diff
@@ -1,124 +1,266 @@
+            assertEquals(getZone.getConnectionDetails().getDashboardUri(), connectionSecret.get("dashboardUrl").asText());
+            assertEquals(getZone.getConnectionDetails().getIssuerId(),connectionSecret.get("issuerId").asText());
+            assertEquals(getZone.getConnectionDetails().getZone().getHttpHeaderName(),connectionSecret.get("zone").get("http-header-name").asText());
+            assertEquals(getZone.getConnectionDetails().getZone().getHttpHeaderValue(),connectionSecret.get("zone").get("http-header-value").asText());
+        } finally {
+            if (isClaimCreated) {
+                deleteZone(claimName, ServiceName.UAA);
+            }
+            if (getZone != null) {
+                Thread.sleep(20000);
+                getZoneByName(zoneName, HttpStatus.NOT_FOUND, "Zone["+namespace+"-" + claimName + "] not found.");
+            }
+        }
+    }
+    // end region Positive Tests
+
+    // region Negative Tests"
+    @Test
+    public void testCreateZoneWithBadInput() throws Exception {
+        String claimName = null;
         boolean isClaimCreated = false;
         try {
-            isClaimCreated = createZone(claimName, null);
-            claim = getZone(claimName);
-            LOGGER.info("Fetched claim: {}", claim);
+            isClaimCreated = createZone(claimName, uaaRequest);
+            Claim claim = null;
+            claim = getZone(claimName, ServiceName.UAA);
+            assertNull("Orchestrator should not create UAA instance for Bad Input", claim);
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains("is invalid: metadata.name: Required value: name or generateName is required"));
         } finally {
             if (isClaimCreated) {
-                deleteZone(claimName);
+                deleteZone(claimName, ServiceName.UAA);
             }
         }
     }
 
-    private boolean createZone(final String claimName, final Object request)
-        throws ServiceInstanceProviderException {
+    @Test
+    public void testGetZoneForNonExisting() throws Exception {
+        String dummyClaimName = (UAA_CLAIM_NAME_PREFIX + "Non_existent_run_id_12345").toLowerCase();
         try {
-            orchestratorUtil.createService(claimName, ServiceName.UAA, OBJECT_MAPPER.valueToTree(request));
-            return true;
+            Claim claim = getZone(dummyClaimName, ServiceName.UAA);
+            assertNull("Orchestrator should return null for Non existing zone", claim);
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains("Service claim " + dummyClaimName + " does not exist"));
+        }
+    }
+
+    @Test
+    public void testDeleteZoneForNonExisting() throws Exception {
+        String dummyClaimName = (UAA_CLAIM_NAME_PREFIX + "Non_existent_run_id_12345").toLowerCase();
+        try {
+            deleteZone(dummyClaimName, ServiceName.UAA);
         } catch (ServiceInstanceProviderException e) {
```

Comment:
change ServiceInstanceProviderException  to Exception

Reply:
Done